src/test/ui/parser/issue-48137-macros-cannot-interpolate-impl-items.rs

```diff
@@ -0,0 +1,34 @@
+// check-pass
+
+fn main() {}
+
+macro_rules! mac_impl {
+    ($i:item) => {
+        struct S;
+        impl S { $i }
+    }
+}
+
+mac_impl! {
+    fn foo() {}
+}
+
+macro_rules! mac_trait {
+    ($i:item) => {
+        trait T { $i }
+    }
+}
+
+mac_trait! {
+    fn foo() {}
+}
+
+macro_rules! mac_extern {
+    ($i:item) => {
+        extern "C" { $i }
+    }
+}
+
+mac_extern! {
+    fn foo();
+}
```

Comment:
Could you add some failing test cases for interpolated items being put into inappropriate containers?

Reply:
By this you mean for example a `struct S;` inside one of the nested contexts?

Reply:
Yes, something like that.

Reply:
Added another test. :)